app/src/main/java/com/libs/globaldata/GlobalData.java

```diff
@@ -49,4 +54,11 @@ public static void clearModel(String modelName, Context context) {
             }
         }
     }
+
+    public static SharedPreferences getUserSettings(Context context){
+        if(userSettings == null){

```

Comment:
It should probably be synchronized just like above methods, in case of multithreading